src/transports/ipc.rs

```diff
@@ -0,0 +1,376 @@
+//! IPC Transport for *nix
+
+extern crate tokio_core;
+extern crate tokio_uds;
+
+use std::thread;
+use std::sync::{self, atomic, Arc};
+use std::collections::BTreeMap;
+use std::io::{self, Read, Write};
+use std::path::Path;
+
+use self::tokio_core::reactor;
+use self::tokio_core::io::{ReadHalf, WriteHalf, Io};
+use self::tokio_uds::UnixStream;
+
+use futures::{self, Sink, Stream, Future, BoxFuture};
+use futures::sync::{oneshot, mpsc};
+use helpers;
+use rpc::{self, Value as RpcValue};
+use {Transport, Error as RpcError};
+
+macro_rules! try_nb {
+  ($e:expr) => (match $e {
+    Ok(t) => t,
+    Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+      return Ok(futures::Async::NotReady)
+    }
+    Err(e) => {
+      warn!("Error: {:?}", e);
+      return Err(())
+    },
+  })
+}
+
+/// Error returned while initializing IPC transport
+#[derive(Debug)]
+pub enum IpcError {
+  /// Wrapped IO error
+  Io(io::Error),
```

Comment:
might as well just return an `io::Error` from the initialization if it's the only variant. this will make integration easier.